bin/fillTestData/Experiences.js

```diff
@@ -0,0 +1,342 @@
+
+  created: () => util.addDays(Date.now(), -_.random(14)),
+
+  feedbackPublic: () => faker.lorem.sentences(),
+};
+
+/**
+ * This method generates an {nUsers} x {nUsers} matrix experience sharing matrix M.
+ * M[i][j] === true if and only if an experience of user[j] shared with user[i] is to be added to the db.
+ * In other words, the array M[i] defines experiences to be present in the profile of User i.
+ *
+ * To cover different cases, the generation is done as follows:
+ * 1. Make sure there are profiles with no experiences,
+ * one experience with and without reply, and
+ * profiles with only one non-public experience.
+ * 2. Populate one profile with as many experiences as possible.
+ * 3. Populate the rest of the profiles randomly according to the specified
+ * {averageExpNumber} and {replyRate}.
+ *
+ * @param {number} nUsers - the number of users in the db
+ * @param {number} averageExpNumber - the average number of references a randomly populated profile has
+ * @param {number} replyRate - the probability for a randomly populated experience to have a reply
+ * @returns {*[]} user experience sharing matrix
+ */
+function generateExperienceSharingMatrix(nUsers, averageExpNumber, replyRate) {
+  const matrix = Array(nUsers)
+    .fill()
+    .map(() => Array(nUsers).fill());
+
+  // cannot share experience with myself
+  Array.from({ length: nUsers }, (x, i) => (matrix[i][i] = null));
+
+  prePopulate(matrix);
+  populateOneProfileWithLotsOfExperiences(matrix, replyRate);
+  populateRandomly(matrix, averageExpNumber, replyRate);
+  return matrix;
+
+  function prePopulate(expMatrix) {
+    const fromSameBatch = function differsOnlyLastDigit(x, y) {
+      return Math.floor(x / 10) === Math.floor(y / 10);
+    };
+    const pt = profileType;
+
+    for (let to = 0; to < expMatrix.length; to++) {
+      for (let from = 0; from < expMatrix.length; from++) {
+        if (
+          fromSameBatch(from, to) &&
+          ((pt.isWriter(from) && pt.isRecipient(to)) ||
+            (pt.isReplier(from) && pt.isWriter(to)))
+        ) {
+          expMatrix[to][from] = true;
+        } else if (pt.isPrePopulated(to) || pt.isPrePopulated(from)) {
+          expMatrix[to][from] = false;
+        }
+      }
+    }
+  }
+
+  function populateOneProfileWithLotsOfExperiences(expMatrix, replyRate) {
+    const profileWithManyExperiences = profileType.getSeqNumWithManyExperiences();
+    for (let from = 0; from < expMatrix.length; from++) {
+      if (
+        profileWithManyExperiences !== from &&
+        !profileType.isPrePopulated(from)
+      ) {
+        const shouldReply = Math.random() < replyRate;
+        expMatrix[profileWithManyExperiences][from] = true;
+        expMatrix[from][profileWithManyExperiences] = shouldReply;
+      }
+    }
+  }
+
+  function populateRandomly(expMatrix, averageExpNumber, replyRate) {
+    function getExperienceSharingRate(nUsers, averageExpNumber, replyRate) {
```

Comment:
Also would be glad to receive any critique regarding the way I structure the JS code, for example, regarding function nesting like this.

Reply:
It does affect readability in my opinion but it's not a biggie in a script like this. I'd be inclined to leave as-is since it works. :-)